### PR TITLE
Add symlink config linter

### LINKED ISSUE #472: Add linter for symlink config files
It would be useful to lint the symlink config file for missing root_links earlier in the CI pipeline.

### PR DESCRIPTION
**Pull request by jonathan-eq**
Resolves #472

### CODE REVIEW

**Comment by jonathan-eq** [comment]
I have tested that this works manually.

File: komodo/lint_symlink_config.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    )
+    parser.add_argument(
+        "config",
+        type=Path,
+        help="a json file describing symlink structure",
+    )
+    return parser.parse_args()
+
+
+def _get_root_nodes(link_dict: Mapping[str, Union[Mapping, str]]) -> Set[str]:
```
Comment (berland) [inline komodo/lint_symlink_config.py:21]: Code duplication with sanity_check.py. Can you use that directly or put it somewhere common?

File: komodo/lint_symlink_config.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    keys = set(link_dict["links"].keys())
+    values = set(link_dict["links"].values())
+    return keys.difference(values)
+
+
+def lint_symlink_config(link_dict: Mapping[str, Union[Mapping, str]]):
```
Comment (berland) [inline komodo/lint_symlink_config.py:27]: Refactor with sanity_check such that we have less duplicated lines?

**Review by berland — APPROVED** [review]
👍🏻

**Pull request merged**